py/lexer.c

```diff
@@ -119,6 +125,37 @@ STATIC bool is_tail_of_identifier(mp_lexer_t *lex) {
     return is_head_of_identifier(lex) || is_digit(lex);
 }
 
+STATIC void swap_char_banks(mp_lexer_t *lex) {
+    if (lex->vstr_postfix_processing) {
+        unichar h0, h1, h2;
+
+        h0 = lex->chr0;
+        h1 = lex->chr1;
+        h2 = lex->chr2;
+
+        lex->chr0 = lex->vstr_postfix.len > 0 ? lex->vstr_postfix.buf[0] : 0;
+        lex->chr1 = lex->vstr_postfix.len > 1 ? lex->vstr_postfix.buf[1] : 0;
+        lex->chr2 = lex->vstr_postfix.len > 2 ? lex->vstr_postfix.buf[2] : 0;
```

Comment:
This was my own form of defensiveness (the number of times I've segfaulted C by not guarding things like this is... embarrassing at best). In this PR, `vstr_postfix` will always have *at least* `.format(x)` or more. If this postfix is ever used for something else (I'm not sure what else you'd want to syntax transform at a lexer level, but who knows), this defensiveness may be useful, otherwise, happy to scrap it for simplicity.

Reply:
Please do remove the unnecessary bits.  Code should be as tight as possible (to save code space and be efficient).  We have very good tests, coverage and CI so any changes that violate assumptions should be picked up.